test_runner/regress/test_remote_storage.py

```diff
@@ -637,4 +621,101 @@ def assert_compacted_and_uploads_queued():
     time.sleep(10)
 
 
+# Branches off a root branch, but does not write anything to the new branch, so it has a metadata file only.
+# Ensures that such branch is still persisted on the remote storage, and can be restored during tenant (re)attach.
+@pytest.mark.parametrize("remote_storage_kind", [RemoteStorageKind.LOCAL_FS])
+def test_created_branch_remote_storage_upload(
+    neon_env_builder: NeonEnvBuilder,
+    remote_storage_kind: RemoteStorageKind,
+):
+    neon_env_builder.enable_remote_storage(
+        remote_storage_kind=remote_storage_kind,
+        test_name="test_created_branch_remote_storage_upload",
+    )
+
+    env = neon_env_builder.init_start()
```

Comment:
I wonder if gc and compaction should be disabled here; they could be not yet started at the time when queued count is checked but start right after.

Reply:
Well, I guess there's actually nothing to gc or compact. Resolving.